File: simplerpc-test/src/main/java/com/jelipo/rpctest/ProxyInterface.java
package com.jelipo.rpctest;

import com.jelipo.simplerpc.annotations.Rpc;
import com.jelipo.simplerpc.annotations.RpcConfig;

import java.util.concurrent.CompletableFuture;

/**
 * @author Jelipo
 * @date 2019/12/20 0:46
 */
@Rpc //在RPC接口上添加@Rpc注解
public interface ProxyInterface {

    @RpcConfig(timeout = 1)
    String getUserData(People people);

    CompletableFuture<People> getUserDataAsysn(String info);
}

File: simplerpc/src/main/kotlin/com/springmarker/simplerpc/client/ProxyInterface.kt
package com.springmarker.simplerpc.client

/**
 * 此接口用于定义如何发送消息，主要定义
 *
 * @author Frank
 * @date 2018/10/16 22:29
 */
interface ProxyInterface {

    fun convertAndSend(any:Any)
}